sample-app/src/components/VisualAutocomplete/VisualSearchBar.tsx

```diff
@@ -50,10 +57,27 @@ export default function VisualSearchBar({
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input) ?? '';
   const isLoading = useAnswersState(state => state.searchStatus.isLoading) ?? false;
-  const [executeQuery, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
+  const [executeQueryWithNearMeHandling, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
   const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(async () => {
     return answersActions.executeUniversalAutocomplete();
   });
+  const recentSearches = useMemo(() => {
+    if (hideRecentSearches) {
+      localStorage.removeItem(DEFAULT_RECENT_SEARCH_KEY);
```

Comment:
I think we probably don't want to put side effects inside useMemo. Is it necessary to remove recent searches data from local storage?

Reply:
wondering if maybe it makes sense to use session storage, maybe rose would know better how long we want recent searches to persist?

Reply:
In the case the user switch hideRecentSearches from false to true, recent searches may no longer track new queries but the old ones still exists in local storage. Based on the spec regarding privacy concerns, we should clear those data. And when hideRecentSearches is switch back to false, they would have a fresh history. I could pull the remove call out of useMemo

Reply:
sgtm!